RearrangeCarGroup8Movements.py

```diff
@@ -0,0 +1,212 @@
+and print a series of moves from position x to y
+
+For our example, the series of moves should be e.g.:
+    move 1 to 2
+    move 3 to 0
+'''
+
+
+class Movement(object):
+    """This class stores information about one movement,
+    i.e. a car moving from position *from_index* to position *to_index*.
+    """
+
+    def __init__(self, car, from_index, to_index):
+        self.car = car
+        self.from_index = from_index
+        self.to_index = to_index
+
+    def __str__(self):
+        move_str = "Move car #{0} from {1} to {2}".format(self.car, self.from_index, self.to_index)
+        return move_str
+
+
+class MovementPrinter(object):
+    """
+    Class used to print the list of movements into stdout or log file.
+    """
+
+    def __init__(self, init_order, final_order, print_console):
+        self.order_message = self._prepare_order_message(init_order, final_order)
+        self.print_console = print_console
+
+    def _prepare_order_message(self, initial_car_order, final_car_order):
+        return ('Initial car order: {init_order}'.format(init_order=initial_car_order) +
+                'Final car order: {final_order}'.format(final_order=final_car_order) +
+                'Moves:')
+
+    def print_moves(self, moves):
+        if len(moves) > 1000:
+            self.print_moves_log_file(moves)
+        else:
+            self.print_moves_console(moves)
+
+    def print_moves_log_file(self, moves_list):
+
+        '''Prints a list of moves to the log file
+
+        Arguments:
+            moves: a list of Movement class instances
+        Returns:
+            None
+        '''
+
+        info_message = 'Number moves is {N} > 1000, logging them..'.format(N=len(moves_list))
+        print(info_message)
+        logging.basicConfig(filename='RearrangeCarGroup8.log', filemode='w', level=logging.DEBUG)
+        logging.info(self.order_message)
+        for move in moves_list:
+            logging.info(move.__str__())
+
+    def print_moves_console(self, moves_list):
+
+        '''Prints a list of moves
+        Arguments:
+            moves: a list of Movement class instances
+        Returns:
+            None
+        '''
+
+        print(self.order_message)
+
+        for move in moves_list:
+            if self.print_console:
+                print(move)
+            else:
+                logging.info(move.__str__())
+        print('\n')
+
+
+class RearrangeCars(object):
+    '''Class to rearrange the cars and print the moves
+       - Uses car_position_dict to optimize searching the position of a given car
+       - Stores additionally the position of empty spot in current car arrangement
+       self.current_car_order.
+       - Moves are either printed to the console or logged
+       - Print moves in O(n), where n is the number of cars.
+       - More precisely at most n + k moves are taken, where k << n is the amount of movements
+       - where car is placed not to the final place

```

Comment:
Don't understand what you mean with "the amount of movements where car is placed not to the final place"